core/domain/question_services.py

```diff
@@ -305,14 +306,20 @@ def delete_question(
             still retained in the datastore. This last option is the preferred
             one.
     """
-    question_model = question_models.QuestionModel.get(question_id)
-    question_model.delete(
+    question_model = question_models.QuestionModel.get_by_id(question_id)
+    if question_model is not None:
+        opportunity_services.increment_question_counts(
+            question_model.linked_skill_ids, -1)
+
+    transaction_services.run_in_transaction(
+        question_models.QuestionModel.delete_multi, [question_id],
```

Comment:
What I meant here is that `opportunity_services.increment_question_counts` and `question_models.QuestionModel.delete_multi` need to be run in one transaction. So put them to one function (preferably inner function) and then run the function in a transaction.

Reply:
I have put those two in an inner function and ran it in a transaction but I am still not sure if it is right and bit confused as to why is it necessary. Can you please explain me. Thanks.

Reply:
Basically, the transaction makes sure that either no operation is performed or all operations are performed. This is needed here since if only `increment_question_counts` is done and then the `delete_multi` fails for some reason the next time we would try to delete the question `increment_question_counts` would be run again meaning that for one deleted question the count would be decremented twice. 

One link for the transactions https://cloud.google.com/appengine/docs/standard/python/ndb/transactions

Reply:
Thank you for the explanation.